pkg/kafsar/group_coordinator.go

```diff
@@ -24,29 +24,32 @@ import (
 )
 
 type Group struct {
-	groupId           string
-	partitionedTopic  []string
-	groupStatus       GroupStatus
-	supportedProtocol string
-	groupProtocols    map[string]string
-	protocolType      string
-	leader            string
-	members           map[string]*memberMetadata
-	canRebalance      bool
-	generationId      int
-	groupLock         sync.RWMutex
-	groupStatusLock   sync.RWMutex
-	groupMemberLock   sync.RWMutex
-	sessionTimeoutMs  int
+	groupId            string
+	partitionedTopic   []string
+	groupStatus        GroupStatus
+	supportedProtocol  string
+	groupProtocols     map[string]string
+	protocolType       string
+	leader             string
+	members            map[string]*memberMetadata
+	canRebalance       bool
+	generationId       int
+	groupLock          sync.RWMutex
+	groupStatusLock    sync.RWMutex
+	groupMemberLock    sync.RWMutex
+	groupNewMemberLock sync.RWMutex
```

Comment:
need group new member lock?

Reply:
The waiting time for rebalancing when a new member is added is relatively long.So need a new lock to avoid lock conflicts.